project/gen2.scala

```diff
@@ -120,6 +120,12 @@ object gen2 {
         io.transact(xa).unsafePerformIO
       },
 
+      enum("F2WindowCover") {
+        type F2WindowCoverRec = Record.`'tag -> String, 'shortName -> String, 'longName -> String, 'tccValue -> String`.T
+        val io = sql"select id, id tag, short_name, long_name, tcc_value from e_f2_window_cover".query[(String, F2WindowCoverRec)].list
+        io.transact(xa).unsafePerformIO
+      },
```

Comment:
Shapeless uses this trick to allow you to express types that you can construct but can't write down in Scala. For instance:

```scala
scala> val a = 1 // Int
a: Int = 1

scala> final val a = 1 // singleton type 1
a: Int(1) = 1

scala> val a: 1 = 1 // but you can't do this (yet)
<console>:1: error: identifier expected but integer literal found.
val a: 1 = 1
       ^

scala> val a: Witness.`1`.T = 1 // so shapeless lets you do this
a: Int(1) = 1
```